### PR TITLE
Accessor API for Remapping

### LINKED ISSUE #629: Accessor API for Remapping
### Proposed new feature or change:

Use an accessor-based implementation for remapping, similar to our `.plot` and `subset` functionality

Instead of:
```Python
uxds.nearest_neighbor_remap()
```

We should employ:
```Python
uxds.remap.nearest_neighbor()
```

### PR DESCRIPTION
**Pull request by aaronzedwick**
Closes #629

## Overview

Changes our current remapping function calls to use an accessor API for remapping.
## Expected Usage

```Python
import uxarray as ux

grid_path = "/path/to/grid.nc"
data_path = "/path/to/data.nc"

uxds = ux.open_dataset(grid_path, data_path)
uxgrid = ux.open_grid(grid_path)

uxds.remap.nearest_neighbor(destination_obj=uxgrid, remap_to="nodes")

```

## PR Checklist

**General**
- [x] An issue is linked created and linked
- [x] Add appropriate labels
- [x] Filled out Overview and Expected Usage (if applicable) sections

**Testing**
- [x] Adequate tests are created if there is new functionality
- [x] Tests cover all possible logical paths in your function
- [x] Tests are not too basic (such as simply calling a function and nothing else)

**Documentation**
- [x] Docstrings have been added to all new functions
- [x] Docstrings have updated with any function changes
- [x] Internal functions have a preceding underscore (`_`) and have been added to `docs/internal_api/index.rst`
- [x] User functions have been added to `docs/user_api/index.rst`

### CODE REVIEW

File: uxarray/remap/dataarray_accessor.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+)
+
+if TYPE_CHECKING:
+    from uxarray.core.dataset import UxDataset
+    from uxarray.core.dataarray import UxDataArray
+
+from uxarray.grid import Grid
+
+
+class UXDataArrayRemapAccessor:
```
Comment (philipc2) [inline uxarray/remap/dataarray_accessor.py:16]: Can we add a `__repr__` like our `subset` accessor?

https://github.com/UXARRAY/uxarray/blob/b6bb18b730eb64ebf5f47e7316247feea27b8ae7/uxarray/subset/dataarray_accessor.py#L18-L28

File: uxarray/remap/dataset_accessor.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+)
+
+if TYPE_CHECKING:
+    from uxarray.core.dataset import UxDataset
+    from uxarray.core.dataarray import UxDataArray
+
+from uxarray.grid import Grid
+
+
+class UXDatasetRemapAccessor:
```
Comment (philipc2) [inline uxarray/remap/dataset_accessor.py:16]: Same about `__repr__`

https://github.com/UXARRAY/uxarray/blob/b6bb18b730eb64ebf5f47e7316247feea27b8ae7/uxarray/subset/dataarray_accessor.py#L18-L28

File: uxarray/remap/dataarray_accessor.py (hunk 1 above)
Comment (philipc2) [inline uxarray/remap/dataarray_accessor.py:16]:
```suggestion
class UxDataArrayRemapAccessor:
```

File: uxarray/remap/dataset_accessor.py (hunk 2 above)
Comment (philipc2) [inline uxarray/remap/dataset_accessor.py:16]:
```suggestion
class UxDatasetRemapAccessor:
```

File: uxarray/core/dataset.py [hunk 3]
```diff
@@ -347,7 +345,13 @@ def nearest_neighbor_remap(
         coord_type : str, default="spherical"
             Indicates whether to remap using on spherical or cartesian coordinates
         """
-        return _nearest_neighbor_uxds(self, destination_obj, remap_to, coord_type)
+        warn(
+            "This implementation of using remapping is being removed in the near future. It is advised to use "
+            "'uxds.remap.nearest_neighbor'",
```
Comment (philipc2) [inline uxarray/core/dataset.py:350]: "This implementation of performing remapping will be deprecated in a future release. It is advised to use `uxds.remap.nearest_neighbor()` instead."

Small rephrasing.

File: uxarray/core/dataset.py [hunk 4]
```diff
@@ -374,6 +378,12 @@ def inverse_distance_weighted_remap(
         k : int, default=8
             Number of nearest neighbors to consider in the weighted calculation.
         """
-        return _inverse_distance_weighted_remap_uxds(
-            self, destination_obj, remap_to, coord_type, power, k
+        warn(
+            "This implementation of using remapping is being removed in the near future. It is advised to use "
+            "'uxds.remap.inverse_distance_weighted'",
+            DeprecationWarning,
+        )
```
Comment (philipc2) [inline uxarray/core/dataset.py:385]: Same rephrasing as above

File: uxarray/core/dataarray.py [hunk 5]
```diff
@@ -298,9 +302,14 @@ def inverse_distance_weighted_remap(
         k : int, default=8
             Number of nearest neighbors to consider in the weighted calculation.
         """
+        warn(
```
Comment (philipc2) [inline uxarray/core/dataarray.py:305]: Same rephrasing as other comment.

File: uxarray/core/dataarray.py [hunk 6]
```diff
@@ -270,8 +269,13 @@ def nearest_neighbor_remap(
         coord_type : str, default="spherical"
             Indicates whether to remap using on spherical or cartesian coordinates
         """
+        warn(
```
Comment (philipc2) [inline uxarray/core/dataarray.py:272]: Same rephrasing as other comment

**Review by philipc2 — CHANGES_REQUESTED** [review]
Nice work, implementation looks good, just a few suggestions.

File: uxarray/remap/dataarray_accessor.py (hunk 1 above)
Comment (aaronzedwick) [reply uxarray/remap/dataarray_accessor.py:16]: Right I forgot about this, sure thing!

**Review by philipc2 — APPROVED** [review]
This looks great! Thanks for putting it together!

**Pull request merged**